docs/DeveloperGuide.md

```diff
@@ -632,40 +661,265 @@ testers are expected to do more *exploratory* testing.
+   
+   3. Test case: `cedit 1 p/91231231`<br>
+      Expected: The first client's phone number will be edited from "99887766" to "91231231". A command success message is displayed in the result display box.
+   
+   4. Test case: `cedit 1 e/peterpan@example.com`<br>
+      Expected: The first client's email address will be edited from "johndoe@example.com" to "peterpan@example.com". A command success message is displayed in the result display box.
+   
+   5. Test case: `cedit 1 n/Peter Wang p/ e/`
+      Expected: No client's information will be edited as a client must at least have a phone number or an email address. Error details will be displayed in the result display box.
+   
+   6. Test case: `cedit 0`
+      Expected: No client's information will be edited. Error details will be displayed in the result display box.
+   
+   7. Test case: `cedit x`, where x is larger than the total number of clients.<br>
+      Expected: No client's information will be edited. Error details will be displayed in the result display box.
+   
+   8. Test case: `cedit`
+      Expected: No client's information will be edited. An invalid command will be shown in the result display box.
+  
+   
+   
+
+#### Editing a client's [_rental information_](#glossary-rental-information)
+
+1. TODO!! @Song Lin
+
+#### Listing all [_clients_](#glossary-client)
+
+1. Listing all _clients_ within the application.
+
+   1. Prerequisites: There are already multiple _clients_ within the application.
+   
+   2. Test case: `list` 
+      Expected: All _clients_ will be displayed in the client list panel.
+
+
+#### Find a keyword within all [_clients_](#glossary-client) and [_rental information_](#glossary-rental-information)
+
+1. Finding a keyword from all _clients_ and _rental information_.
+
+    1. Prerequisites: List all _clients_ using the `list` command. Multiple _clients_ in the list, with client called Amy Tan in this list.
+
+    2. Test case: `find k/Amy Tan`<br>
+       Expected: Client with the keyword "Amy Tan" will be displayed on the client list panel. Command success message will be displayed in the result display box.
+
+    3. Test case: `find k/Yong Li`<br>, given that no client within the list is called "Yong Li"
+       Expected: No client or rental information matches the particular keyword "Yong Li". Error details will be displayed in the result display box.
+
+    4. Test case: `find Amy`<br>
+       Expected: No prefix k/ given. An invalid command will be shown in the result display box.
+
+    5. Test case: `find k/`<br>
+       Expected: No parameters given. An invalid command will be shown in the result display box.
+
+    6. Test case: `find`<br>
+       Expected: No prefix and/or parameters given. An invalid command will be shown in the result display box.
+
+#### Find a [_clients_](#glossary-client)
+
+1. TODO!! @Nathan
```

Comment:
may be good to put this as comment and the text displaying change to under development or something, same apply for rest of the TODO